pkg/monitor/monitor/monitor_manager.go

```diff
@@ -147,71 +137,96 @@ func (m *MonitorManager) SyncMonitor(monitor *v1alpha1.TidbMonitor) error {
 
 func (m *MonitorManager) syncTidbMonitorService(monitor *v1alpha1.TidbMonitor) error {
 	services := getMonitorService(monitor)
-	for _, svc := range services {
-		_, err := m.deps.TypedControl.CreateOrUpdateService(monitor, svc)
+	for _, newSvc := range services {
+		oldSvc, err := m.deps.ServiceLister.Services(newSvc.Namespace).Get(newSvc.Name)
+		if errors.IsNotFound(err) {
+			err = controller.SetServiceLastAppliedConfigAnnotation(newSvc)
+			if err != nil {
+				return err
+			}
+			err = m.deps.ServiceControl.CreateService(monitor, newSvc)
+			if err != nil {
+				return err
+			}
+			continue
+		}
 		if err != nil {
-			klog.Errorf("tm[%s/%s]'s service[%s] failed to sync,err: %v", monitor.Namespace, monitor.Name, svc.Name, err)
-			return controller.RequeueErrorf("tm[%s/%s]'s service[%s] failed to sync,err: %v", monitor.Namespace, monitor.Name, svc.Name, err)
+			return fmt.Errorf("syncTidbMonitorService: failed to get svc %s for cluster %s/%s, error: %s", newSvc.Name, monitor.Namespace, monitor.Name, err)
 		}
-	}
-	return nil
-}
 
-func (m *MonitorManager) syncTidbMonitorPVC(monitor *v1alpha1.TidbMonitor) (*corev1.PersistentVolumeClaim, error) {
-
-	pvc := getMonitorPVC(monitor)
-	pvc, err := m.deps.TypedControl.CreateOrUpdatePVC(monitor, pvc, false)
-	if err != nil {
-		klog.Errorf("tm[%s/%s]'s pvc failed to sync,err: %v", monitor.Namespace, monitor.Name, err)
-		return nil, err
-	}
-	return pvc, nil
-}
+		equal, err := controller.ServiceEqual(newSvc, oldSvc)
```

Comment:
We have to preserve some fields in the service spec, e.g. clusterIP, NodePort, etc., please refer to the function for TiDB https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/tidb_member_manager.go#L256.
Please also check the logs of `tidb-controller-manager` during your test to make sure that no unnecessary updates for the resources during the sync.